libntfs-3g/inode.c

```diff
@@ -1011,6 +1011,7 @@ int ntfs_inode_sync_in_dir(ntfs_inode *ni, ntfs_inode *dir_ni)
 
 	/* Update STANDARD_INFORMATION. */
 	if ((ni->mrec->flags & MFT_RECORD_IN_USE) && ni->nr_extents != -1 &&
+			NInoDirty(ni) &&
```

Comment:
normally this function called when inode is dirty.
but it called in ntfsck directly for memory leak of nidata cache when trying to open of opened inode.
so I'll look into the code of calling this function in ntfsck